numpyro/distributions/mixtures.py

```diff
@@ -389,9 +397,19 @@ def component_sample(self, key, sample_shape=()):
         return jnp.stack(samples, axis=self.mixture_dim)
 
     def component_log_probs(self, value):
-        component_log_probs = jnp.stack(
-            [d.log_prob(value) for d in self.component_distributions], axis=-1
-        )
+        if self._validate_args:
+            component_log_probs = []
+            for d in self.component_distributions:
+                log_prob = d.log_prob(value)
+                if (self._support is not None) and d._validate_args:
```

Comment:
if `d._validate_args` is True, `log_prob` is already masked so we can skip the masking logic here.

Reply:
Make sense.